### PR TITLE
[Tests-Only][CI] fix the starlark code for reduced ci after merge and sonarcloud

### PR DESCRIPTION
**Pull request by dpakach**
fix the starlark code for reduced ci after merge and sonarcloud

https://github.com/owncloud/QA/issues/649

### CODE REVIEW

**Review by phil-davis — CHANGES_REQUESTED** [review]
This repo has `.drone.yml` - we need to generate that from this `.drone.starlark`
But:
```
$ drone starlark --source .drone.starlark 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7bdb21]

goroutine 1 [running]:
github.com/drone/drone-yaml/yaml/pretty.printEnviron(0xc7da60, 0xc000c8d7c0, 0xc000358c30)
	/go/pkg/mod/github.com/drone/drone-yaml@v0.0.0-20190729072335-70fa398b3560/yaml/pretty/container.go:127 +0x261
github.com/drone/drone-yaml/yaml/pretty.printContainer(0xc7da60, 0xc000c8d7c0, 0xc000100800)
	/go/pkg/mod/github.com/drone/drone-yaml@v0.0.0-20190729072335-70fa398b3560/yaml/pretty/container.go:52 +0xa45
github.com/drone/drone-yaml/yaml/pretty.printPipeline(0xc7da00, 0xc000f138f0, 0xc0002f5880)
	/go/pkg/mod/github.com/drone/drone-yaml@v0.0.0-20190729072335-70fa398b3560/yaml/pretty/pipeline.go:54 +0x90d
github.com/drone/drone-yaml/yaml/pretty.Print(0xc68da0, 0xc000440090, 0xc00009b280)
	/go/pkg/mod/github.com/drone/drone-yaml@v0.0.0-20190729072335-70fa398b3560/yaml/pretty/pretty.go:35 +0x2de
github.com/drone/drone-cli/drone/starlark.generate(0xc000364160, 0xb115c0, 0x0)
	/drone/src/drone/starlark/starlark.go:200 +0xea8
github.com/drone/drone-cli/drone/starlark.glob..func1(0xc000364160)
	/drone/src/drone/starlark/starlark.go:26 +0x40
github.com/urfave/cli.HandleAction(0xa606e0, 0xb96160, 0xc000364160, 0xc000300200, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.20.0/app.go:492 +0x7c
github.com/urfave/cli.Command.Run(0xb5e411, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0xb730a7, 0x21, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.20.0/command.go:210 +0x991
github.com/urfave/cli.(*App).Run(0xc000338340, 0xc0000ac040, 0x4, 0x4, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.20.0/app.go:255 +0x6ab
main.main()
	/drone/src/drone/main.go:100 +0x93f
```

So we need to sort out what goes wrong with generating `.drone.yml`

File: .drone.starlark [hunk 1]
```diff
@@ -1640,147 +1692,226 @@ def acceptance():
...
+				'name': 'sonarcloud',
+				'image': 'sonarsource/sonar-scanner-cli',
+				'pull': 'always',
+				'environment': {
+					'SONAR_TOKEN': {
+						'from_secret': 'sonar_token'
+					},
+					'SONAR_PULL_REQUEST_BASE': 'master' if ctx.build.event == 'pull_request' else None,
+					'SONAR_PULL_REQUEST_BRANCH': ctx.build.source if ctx.build.event == 'pull_request' else None,
+					'SONAR_PULL_REQUEST_KEY': ctx.build.ref.replace("refs/pull/", "").split("/")[0] if ctx.build.event == 'pull_request' else None,
```
Comment (dpakach) [inline .drone.starlark:1878]: This is the line causing the error in starlark generate step @phil-davis

**Comment by dpakach** [comment]
Ok, apparently `drone starlark generate` fails if we set env variables to `None`. Changing them to empty strings seems to have fixed the issue here, @phil-davis do we need to fix that on all other apps.
We only use `drone starlark generate` on this app only.

**Comment by phil-davis** [comment]
Make the change here, then make a PR to `activity` app to make the change there also. Then the next time we have important changes to the "standard" starlark, this will get propagated to the other oC10 apps.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**